common/src/main/java/tachyon/ClientBase.java

```diff
@@ -80,10 +90,28 @@ public ClientBase(InetSocketAddress address, TachyonConf tachyonConf, String mod
   protected abstract String getServiceName();
 
   /**
+   * Checks that the service version is compatible with the client.
+   */
+  protected void checkVersion(TachyonService.Client client, long version) throws IOException {
+    if (mServiceVersion == Constants.UNKNOWN_SERVICE_VERSION) {
+      try {
+        mServiceVersion = client.getServiceVersion();
+      } catch (TException e) {
+        throw new IOException(e.getMessage());
+      }
+      if (mServiceVersion != Constants.BLOCK_MASTER_SERVICE_VERSION) {
+        throw new IOException(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(version,
```

Comment:
Would it be useful to also print out the service name that has the incompatible version?